### PR TITLE
Sphinx + Breathe + Doxygen docs + ReadTheDocs

### LINKED ISSUE #1475: Use sphinx for prettier documentation
## Description
We currently use `doxygen` for C++ and python documentation, but the documentation looks pretty bad. `sphinx` is an alternative documentation generator that looks a lot better.

## Requirements

- Generate documentation using `sphinx` instead of `doxygen`
- Update CI to use the `sphinx` generated documentation instead of the `doxygen` one.

### PR DESCRIPTION
**Pull request by oswinso**
## Description
This PR sets up the skeleton for documentation based off of Sphinx + Breathe + Doxygen.

## Associated Issue
Fixes #1475. 

### Docs render properly.
1. Install new deps: `pip3 install sphinx breathe sphinx_rtd_theme`
2. Go to `build` folder
3. `ninja sphinx`

Expected result: There should be an `index.html` at `build/docs/docs_build/sphinx_docs` that shows the following:
Screenshot_2020-08-17 Introduction — rj_robocup documentation

### ReadTheDocs works properly
1. Go to [https://rj-robocup.readthedocs.io/en/feat-ros2_doc](https://rj-robocup.readthedocs.io/en/feat-ros2_doc/)

Expected result: docs build and render properly.

### CODE REVIEW

File: soccer/src/soccer/gameplay/gameplay_module.hpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ namespace Gameplay {
  * iteration of the main run loop, the GameplayModule
  * calls into python, which does all of the high-level planning, resulting in
  * updated motion targets, etc for the robots.  The
- * GameplayModule then executes path planning for each OurRobot.
+ * GameplayModule then executes path planning for each ::OurRobot.
```
Comment (JNeiger) [inline soccer/src/soccer/gameplay/gameplay_module.hpp]: ?

**Review by JNeiger — APPROVED** [review]

File: soccer/src/soccer/gameplay/gameplay_module.hpp (hunk 1 above)
Comment (oswinso) [reply soccer/src/soccer/gameplay/gameplay_module.hpp]: oh woops I was testing to see why breathe didn't link `OurRobot` so I was messing around. I'll revert this.

**Comment by CollinAvidano** [comment]
What is the purpose of the src folder for docs? Also you all should probably set the depth of toc trees to 1. It makes it so its not just a glob of content in the nav bar. Try it out you will see what I mean. Also ideally the conf.py would be at the same level as the cmakelists.txt and doxyfile. Is there a particular reason this isnt the case?

**Comment by CollinAvidano** [comment]
Also the top level index.rst appears to be an autogenerated one / duplicate of the one in src. If its not in use get rid of it.

**Review by CollinAvidano — APPROVED** [review]
looks good to me.

**Comment by CollinAvidano** [comment]
One thing that I didnt want to block the review on as its pretty minor was just maybe change the project name from rj-robocup to robocup-software for consistency with other repos. But done worry about it too much.

**Comment by oswinso** [comment]
@CollinAvidano the problem with changing the project to `robocup-software` is that to other people, we're RoboJackets, not "robocup software"

**Pull request merged**